utils/httpUtils.ts

```diff
@@ -40,15 +60,35 @@ export const get = async (url: string, args: any, context: any) => {
 export const getFullResponse = async (url: string, args: any, context: any) => {
   try {
     const baseURL = process.env.API_URL;
-    const urlPrefix = args.snapshotLinkId ? `/pub/${args.snapshotLinkId}` : "";
-    const response = await fetch(baseURL + urlPrefix + url, {
-      method: "GET",
-      headers: {
-        Cookie: context.request.headers.get("cookie"),
-        "Content-Type": "application/json",
-      },
-    });
-    return response;
+    const nextGenEnabled = await isNextGenEnabled(context);
+    if (nextGenEnabled) {
+      // next gen details
+      const czidServicesToken = await getEnrichedToken(context);
+      const query = context.params.query;
+      const response = await fetch(process.env.NEXTGEN_ENTITIES_API_URL, {
+        method: "POST",
+        headers: {
+          Cookie: context.request.headers.get("cookie"),
+          "Content-Type": "application/json",
+          "X-CSRF-Token": args?.input?.authenticityToken,
+          Authorization: `Bearer ${czidServicesToken}`,
+        },
+        body: JSON.stringify(query),
+      });
+      return response;
```

Comment:
(reposting [previous comment](https://github.com/chanzuckerberg/czid-graphql-federation-server/pull/44#discussion_r1491386917))

would it be viable to put this code in the `callGraphqlSource` util function instead, and call that util function here?  This util file is getting fairly long and complex, so it'd be nice to extract the chunks that make sense.

https://github.com/chanzuckerberg/czid-graphql-federation-server/blob/010967bd3011dd1544a6bad984f4e64e2f3f0b08/utils/callGraphqlSource.ts#L3

Reply:
I had another thought here, that might be worth adding a comment about.  All the calls that Relay makes to graphQL mesh are done via POST, so it feels a bit weird that within the get method call, we are calling this.  

I understand why that it's necesary, because otherwise we would need to branch in every resolver based on the feature flag, which is not ideal.  But I'm wondering if that's worth a one line comment to explain in case someone is confused in the future.

Reply:
I don't know if this is going to work the way we want because in the custom resolvers we are already massaging the Rails return to fit the PREDICTED NextGen. In the short term, it might be easier to use the feature flag in each resolver to output the ACTUAL NextGen response and map it to what the front end now expects, rather than changing the front end again right now?

What do you think?